Removed CalibTracker/SiStripCommon dependency from CalibFormats/SiStripObjects

Dr15Jones opened the pull request:
#### PR description:

CalibFormat objects should not be reading files to get intermediate data as this violates one of the guiding principles of CMSSW.  This change moves the needed information into a separate class, SiStripDetInfo and requires code that constructs a SiStripQuality or a SiStripGain to pass in the appropriate SiStripDetInfo.

The class SiStripDetInfoFileReader now returns the SiStripDetInfo.

This now makes CalibFormats/SiStripObjects independent from algorithmic code.

#### PR validation:

Code compiles.

Dr15Jones committed:
Removed file reading from SiStripQuality

The information needed is now in SiStripDetInfo. It is up to the
code that constructs a SiStripQuality to get access to a
SiStripDetInfo.

Dr15Jones committed:
Removed file reading from SiStripGain

This allows dropping dependency on CalibTracker/SiStripCommon.

Dr15Jones commented:
@davidlange6 this should help with the ROOT modules build.

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33888/22931

- This PR adds an extra 88KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DQMOffline/CalibTracker/plugins/SiStripQualityHotStripIdentifierRoot.cc  modified in PR(s): #33880

Code check has found code style and quality issues which could be resolved by applying following patch(s)

- **code-checks**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33888/22931/code-checks.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33888/22931/code-checks.patch | patch -p1`
You can also run `scram build code-checks` to apply code checks directly

Dr15Jones committed:
code checks

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33888/22936

- This PR adds an extra 88KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DQMOffline/CalibTracker/plugins/SiStripQualityHotStripIdentifierRoot.cc  modified in PR(s): #33880

cmsbuild commented:
A new Pull Request was created by  @Dr15Jones (Chris Jones) for master.

It involves the following packages:

CalibFormats/SiStripObjects
CalibTracker/SiStripCommon
CalibTracker/SiStripESProducers
CalibTracker/SiStripHitEfficiency
CalibTracker/SiStripQuality
CondCore/SiStripPlugins
DQM/SiStripMonitorSummary
DQMOffline/CalibTracker
OnlineDB/SiStripESSources
RecoLocalTracker/SiStripClusterizer


@perrotta, @malbouis, @andrius-k, @yuanchao, @kmaeshima, @tlampen, @ErnestaP, @ahmad3213, @cmsbuild, @jpata, @jfernan2, @slava77, @ggovi, @pohsun, @rvenditti, @francescobrivio can you please review it and eventually sign? Thanks.
@erikbutz, @echabert, @felicepantaleo, @pieterdavid, @robervalwalsh, @fioriNTU, @threus, @venturia, @hdelanno, @makortel, @JanFSchulte, @rociovilar, @yduhm, @sroychow, @GiacomoSguazzoni, @rovere, @VinInn, @tocheng, @jandrea, @alesaggio, @idebruyn, @mmusich, @mtosi, @arossi83, @gbenelli, @gpetruc this is something you requested to watch as well.
@silviodonato, @dpiparo, @qliphy you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

jfernan2 commented:
please test

cmsbuild commented:
-1

**Failed Tests**:   Build  ClangBuild 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-59ed9d/15406/summary.html
**COMMIT**: fa8eceb40cd9406ba134207dedd02824baa5d9be
**CMSSW**: CMSSW_12_0_X_2021-05-28-2300/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/33888/15406/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Build

I found compilation error when building: 

<pre>>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-05-28-2300/src/CalibTracker/SiStripESProducers/plugins/fake/Phase2TrackerCablingCfgESSource.cc
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-05-28-2300/src/CalibTracker/SiStripESProducers/plugins/fake/SiStripApvGainFakeESSource.cc
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-05-28-2300/src/CalibTracker/SiStripESProducers/plugins/fake/SiStripBackPlaneCorrectionFakeESSource.cc
In file included from /cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc900/cms/cmssw/CMSSW_12_0_X_2021-05-28-2300/src/Geometry/TrackerNumberingBuilder/interface/utils.h:4,
                 from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-05-28-2300/src/CalibTracker/SiStripESProducers/plugins/fake/SiStripBackPlaneCorrectionFakeESSource.cc:26:
/cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc900/cms/cmssw/CMSSW_12_0_X_2021-05-28-2300/src/Geometry/TrackerNumberingBuilder/interface/GeometricDet.h:12:10: fatal error: DD4hep/Shapes.h: No such file or directory
   12 | #include <DD4hep/Shapes.h>
      |          ^~~~~~~~~~~~~~~~~
compilation terminated.
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-05-28-2300/src/CalibTracker/SiStripESProducers/plugins/fake/SiStripBadModuleConfigurableFakeESSource.cc
gmake: *** [tmp/slc7_amd64_gcc900/src/CalibTracker/SiStripESProducers/plugins/CalibTrackerSiStripFakeESProducersPlugins/fake/SiStripBackPlaneCorrectionFakeESSource.cc.o] Error 1
</pre>

## Clang Build

I found compilation error while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 32 COMPILER='llvm compile'

```


<pre>>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-05-28-2300/src/CalibTracker/SiStripESProducers/plugins/fake/SiStripPedestalsFakeESSource.cc
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-05-28-2300/src/CalibTracker/SiStripESProducers/plugins/fake/SiStripQualityFakeESSource.cc
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-05-28-2300/src/CalibTracker/SiStripESProducers/plugins/fake/SiStripThresholdFakeESSource.cc
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-05-28-2300/src/CalibTracker/SiStripESProducers/plugins/fake/modules.cc
In file included from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-05-28-2300/src/CalibTracker/SiStripESProducers/plugins/fake/SiStripHashedDetIdFakeESSource.cc:6:
/cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc900/cms/cmssw/CMSSW_12_0_X_2021-05-28-2300/src/Geometry/TrackerNumberingBuilder/interface/GeometricDet.h:12:10: fatal error: 'DD4hep/Shapes.h' file not found
#include <DD4hep/Shapes.h>
         ^~~~~~~~~~~~~~~~~
In file included from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-05-28-2300/src/CalibTracker/SiStripESProducers/plugins/fake/SiStripLorentzAngleFakeESSource.cc:36:
/cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc900/cms/cmssw/CMSSW_12_0_X_2021-05-28-2300/src/Geometry/TrackerNumberingBuilder/interface/GeometricDet.h:12:10: fatal error: 'DD4hep/Shapes.h' file not found
#include <DD4hep/Shapes.h>
</pre>

Dr15Jones commented:
@smuzaffar why did the `<use name="dd4hep"/>` in Geometry/TrackerNumberingBuilder/BuildFile.xml not get applied to the plugins in CalibTracker/SiStripESProducers?

Dr15Jones committed:
Updated BuildFile.xml files to new dependencies

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33888/22944

- This PR adds an extra 216KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DQMOffline/CalibTracker/plugins/SiStripQualityHotStripIdentifierRoot.cc  modified in PR(s): #33880

Dr15Jones committed:
Removed unneeded include of FileInPath.h

cmsbuild commented:
Pull request #33888 was updated. @perrotta, @malbouis, @civanch, @yuanchao, @kmaeshima, @tlampen, @andrius-k, @mdhildreth, @ErnestaP, @cmsbuild, @jpata, @jfernan2, @ahmad3213, @slava77, @ggovi, @pohsun, @rvenditti, @francescobrivio can you please check and sign again.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33888/22945

- This PR adds an extra 12KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DQMOffline/CalibTracker/plugins/SiStripQualityHotStripIdentifierRoot.cc  modified in PR(s): #33880

Dr15Jones commented:
please test

Dr15Jones committed:
Added missing include of FileInPath.h

Dr15Jones committed:
Removed unused Service API from SiStripDetInfoFileReader

Although it had the constructor like a Service, the class was
never registered to be a Service.

Dr15Jones commented:
I have another round of cleanups on its way.

Dr15Jones committed:
Added SiStripDetInfoFileReader::kDefaultFile

Use the constant in recently changed files.

cmsbuild commented:
-1

**Failed Tests**: UnitTests 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-59ed9d/15412/summary.html
**COMMIT**: bab4b18c542b19a0aab47d4537461d5eacf8bd1a
**CMSSW**: CMSSW_12_0_X_2021-05-28-2300/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/33888/15412/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Unit Tests

I found errors in the following unit tests:

<pre>---> test SiStripDAQ_O2O_test had ERRORS
</pre>

## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 10 differences found in the comparisons
 - DQMHistoTests: Total files compared: 37
 - DQMHistoTests: Total histograms compared: 2650486
 - DQMHistoTests: Total failures: 18
 - DQMHistoTests: Total nulls: 1
 - DQMHistoTests: Total successes: 2650445
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: -0.004 KiB( 36 files compared)
 - DQMHistoSizes: changed ( 312.0 ): -0.004 KiB       MessageLogger/Warnings
 - Checked 155 log files, 37 edm output root files, 37 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_0_X_2021-05-28-2300+59ed9d/43051/triggerResults): no differences found

Dr15Jones commented:
finished with what I wanted to do

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33888/22948

- This PR adds an extra 120KB to repository

Code check has found code style and quality issues which could be resolved by applying following patch(s)

- **code-format**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33888/22948/code-format.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33888/22948/code-format.patch | patch -p1`
You can also run `scram build code-format` to apply code format directly

Dr15Jones committed:
code-format

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33888/22951

- This PR adds an extra 124KB to repository

cmsbuild commented:
Pull request #33888 was updated. @perrotta, @malbouis, @civanch, @yuanchao, @kmaeshima, @tlampen, @andrius-k, @mdhildreth, @ErnestaP, @cmsbuild, @jpata, @jfernan2, @ahmad3213, @slava77, @ggovi, @pohsun, @rvenditti, @francescobrivio can you please check and sign again.

Dr15Jones commented:
Please test

cmsbuild commented:
-1

**Failed Tests**: UnitTests 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-59ed9d/15422/summary.html
**COMMIT**: f589717429f399c81cb0058136e7ed407c066a66
**CMSSW**: CMSSW_12_0_X_2021-05-28-2300/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/33888/15422/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Unit Tests

I found errors in the following unit tests:

<pre>---> test SiStripDAQ_O2O_test had ERRORS
</pre>

## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 6 differences found in the comparisons
 - DQMHistoTests: Total files compared: 37
 - DQMHistoTests: Total histograms compared: 2650486
 - DQMHistoTests: Total failures: 13
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2650451
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 36 files compared)
 - Checked 155 log files, 37 edm output root files, 37 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_0_X_2021-05-28-2300+59ed9d/43055/triggerResults): no differences found

Dr15Jones commented:
please test

cmsbuild commented:
-1

**Failed Tests**: UnitTests 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-59ed9d/15509/summary.html
**COMMIT**: f589717429f399c81cb0058136e7ed407c066a66
**CMSSW**: CMSSW_12_0_X_2021-05-31-2300/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/33888/15509/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Unit Tests

I found errors in the following unit tests:

<pre>---> test SiStripDAQ_O2O_test had ERRORS
</pre>

## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 4 differences found in the comparisons
 - DQMHistoTests: Total files compared: 37
 - DQMHistoTests: Total histograms compared: 2650486
 - DQMHistoTests: Total failures: 6
 - DQMHistoTests: Total nulls: 1
 - DQMHistoTests: Total successes: 2650457
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: -0.004 KiB( 36 files compared)
 - DQMHistoSizes: changed ( 312.0 ): -0.004 KiB       MessageLogger/Warnings
 - Checked 155 log files, 37 edm output root files, 37 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_0_X_2021-05-31-2300+59ed9d/43114/triggerResults): no differences found

Dr15Jones commented:
NOTE: ` test SiStripDAQ_O2O_test had ERRORS` is failing in the IBs as well. Therefore it is not a consequence of this PR.

mmusich commented:
@cmsbuild, please test.
In the last IB the issue with `SiStripDAQ_O2O_test` [is solved](https://cmssdt.cern.ch/SDT/cgi-bin/logreader/slc7_amd64_gcc900/CMSSW_12_0_X_2021-06-02-1100/unitTestLogs/CondTools/SiStrip#/)

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-59ed9d/15595/summary.html
**COMMIT**: f589717429f399c81cb0058136e7ed407c066a66
**CMSSW**: CMSSW_12_0_X_2021-06-02-2300/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/33888/15595/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



The workflows 140.53 have different files in step1_dasquery.log than the ones found in the baseline. You may want to check and retrigger the tests if necessary. You can check it in the "files" directory in the results of the comparisons

**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 1260 differences found in the comparisons
 - DQMHistoTests: Total files compared: 37
 - DQMHistoTests: Total histograms compared: 2650486
 - DQMHistoTests: Total failures: 3669
 - DQMHistoTests: Total nulls: 19
 - DQMHistoTests: Total successes: 2646776
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: -45.703 KiB( 36 files compared)
 - DQMHistoSizes: changed ( 140.53 ): -44.531 KiB      Hcal/DigiRunHarvesting
 - DQMHistoSizes: changed ( 140.53 ): -1.172 KiB       RPC/DCSInfo
 - Checked 155 log files, 37 edm output root files, 37 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_0_X_2021-06-02-2300+59ed9d/43181/triggerResults): no differences found

jfernan2 commented:
+1

slava77 commented on RecoLocalTracker/SiStripZeroSuppression/BuildFile.xml:
```diff
@@ -8,6 +8,7 @@
 <use name="CondFormats/SiStripObjects"/>
 <use name="CalibFormats/SiStripObjects"/>
 <use name="CalibTracker/Records"/>
+<use name="Geometry/TrackerGeometryBuilder"/>
```
why was this needed?
There are no includes of `Geometry/TrackerGeometryBuilder` in ` RecoLocalTracker/SiStripZeroSuppression/`

slava77 replied:
nevermind, I removed the comment; there's no problem here

slava77 commented:
+reconstruction

for https://github.com/cms-sw/cmssw/pull/33888	f589717
- code changes in reco are rather trivial: 
    - adapting to the new interface in RecoLocalTracker/SiStripClusterizer/test
    - update/fix of some missing package dependencies
- jenkins tests pass and comparisons with the baseline show no (relevant) differences

civanch commented:
+1

yuanchao commented:
+1

* remove CalibTracker/SiStripCommon dependency
* matrix tests and unit tests passed

qliphy commented:
kindly ping @cms-sw/db-l2

Dr15Jones commented:
Ping @cms-sw/db-l2

ggovi commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @silviodonato, @dpiparo, @qliphy (and backports should be raised in the release meeting by the corresponding L2)

qliphy commented:
+1
